velox/common/base/SimdUtil-inl.h

```diff
@@ -186,7 +186,7 @@ int32_t indicesOfSetBits(
       row += 64;
       continue;
     }
-    if (wordIndex == firstWord && begin) {
+    if (wordIndex == firstWord && begin != firstWord * 64) {
```

Comment:
I'm actually not sure if these special cases are good for the performance. By slightly modifying the do-while loop below, we can avoid calling trailing zero functions if any word is 0. Adding these if checks may increase the branch misprediction and also makes the function too complex.

Reply:
Can you create a benchmark and the new implementation in a new PR?  There is a code freeze starting this afternoon until Nov 28 and we can check and iterate the new implementation during that.

Reply:
I was just btw-ing. There're many such new PRs to be made. I'll work on them next year after the Parquet reader is functional complete.